### ISSUE TITLE
Failed when Create a new Medusa project

### ISSUE THREAD

**Issue by fitrarhm** [medusajs/medusa#1194]
When trying to install **Medusa**, i got fail & long error message.

This is happen when  i do running this command "**_medusa new my-medusa-store --seed_**"

Node version :  v16.13.2
npm : v (8.1.2)

........
.......
.......
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\MYPC\AppData\Local\npm-cache\_logs\2022-03-16T10_56_28_005Z-debug.log
Error: Command failed with exit code 1: npm install
    at makeError (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\node_modules\execa\lib\error.js:60:11)
    at handlePromise (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\node_modules\execa\index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 1: npm install',
  command: 'npm install',
  escapedCommand: 'npm install',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

**Comment by olivermrbl** [comment]
Thanks for reporting the issue. Did your project get created, or did it fail before reaching that point? @fitrarhm

**Comment by fitrarhm** [comment]
Hi [olivermrbl](https://github.com/olivermrbl). It failed immediately after the command "medusa new my-medusa-store --seed".  I found the folder containing the medusa project, but I doubt if it's complete. And when I try to run **NPM START (medusa develop)**, it still doesn't work with the following error message.

> medusa-starter-default@0.0.1 start
> medusa develop

medusa develop

Start development server. Watches file and rebuilds when something changes

Options:
  --verbose                Turn on verbose output                                             [boolean] [default: false]
  --no-color, --no-colors  Turn off the color in output                                       [boolean] [default: false]
  --json                   Turn on the JSON logger                                            [boolean] [default: false]
  -H, --host               Set host. Defaults to localhost                               [string] [default: "localhost"]
  -p, --port               Set port. Defaults to 9000                                         [string] [default: "9000"]
  -h, --help               Show help                                                                           [boolean]
  -v, --version            Show version number                                                                 [boolean]
TypeError: cmd is not a function
    at C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\dist\create-cli.js:260:7
    at C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\dist\create-cli.js:108:24
    at Object.handler (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\dist\create-cli.js:54:24)
    at Object.runCommand (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\node_modules\yargs\build\lib\command.js:196:48)
    at Object.parseArgs [as _parseArgs] (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\node_modules\yargs\build\lib\yargs.js:1043:55)
    at Object.parse (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\node_modules\yargs\build\lib\yargs.js:584:29)
    at module.exports (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\dist\create-cli.js:392:6)
    at Object.<anonymous> (C:\Users\MYPC\AppData\Roaming\npm\node_modules\@medusajs\medusa-cli\dist\index.js:70:27)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)

**Comment by sweety-cool** [comment]
> Hi [olivermrbl](https://github.com/olivermrbl). It failed immediately after the command "medusa new my-medusa-store --seed". I found the folder containing the medusa project, but I doubt if it's complete. And when I try to run **NPM START (medusa develop)**, it still doesn't work with the following error message.
> 
> > medusa-starter-default@0.0.1 start
> > medusa develop
> 
> medusa develop
> 
> Start development server. Watches file and rebuilds when something changes
> 
> Options: --verbose Turn on verbose output [boolean] [default: false] --no-color, --no-colors Turn off the color in output [boolean] [default: false] --json Turn on the JSON logger [boolean] [default: false] -H, --host Set host. Defaults to localhost [string] [default: "localhost"] -p, --port Set port. Defaults to 9000 [string] [default: "9000"] -h, --help Show help [boolean] -v, --version Show version number [boolean] TypeError: cmd is not a function at C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\dist\create-cli.js:260:7 at C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\dist\create-cli.js:108:24 at Object.handler (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\dist\create-cli.js:54:24) at Object.runCommand (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\node_modules\yargs\build\lib\command.js:196:48) at Object.parseArgs [as _parseArgs] (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\node_modules\yargs\build\lib\yargs.js:1043:55) at Object.parse (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\node_modules\yargs\build\lib\yargs.js:584:29) at module.exports (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\dist\create-cli.js:392:6) at Object. (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\dist\index.js:70:27) at Module._compile (node:internal/modules/cjs/loader:1103:14) at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)

**Comment by sweety-cool** [comment]
> When trying to install **Medusa**, i got fail & long error message.
> 
> This is happen when i do running this command "**_medusa new my-medusa-store --seed_**"
> 
> Node version : v16.13.2 npm : v (8.1.2)
> 
> ........ ....... ....... npm ERR! A complete log of this run can be found in: npm ERR! C:\Users\MYPC\AppData\Local\npm-cache_logs\2022-03-16T10_56_28_005Z-debug.log Error: Command failed with exit code 1: npm install at makeError (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\node_modules\execa\lib\error.js:60:11) at handlePromise (C:\Users\MYPC\AppData\Roaming\npm\node_modules@medusajs\medusa-cli\node_modules\execa\index.js:118:26) at processTicksAndRejections (node:internal/process/task_queues:96:5) { shortMessage: 'Command failed with exit code 1: npm install', command: 'npm install', escapedCommand: 'npm install', exitCode: 1, signal: undefined, signalDescription: undefined, stdout: undefined, stderr: undefined, failed: true, timedOut: false, isCanceled: false, killed: false }

yes I'm facing the same problem when I execute "medusa new my-medusa-store --seed" this command. same problem here

**Comment by olivermrbl** [comment]
@fitrarhm @sksar @sweety-cool It seems to be an issue with the `--seed` argument of the command. Though, the Medusa project should have successfully been created, so you can navigate to the folder and run `yarn seed` as a workaround for now.

I'll investigate the issue. Thanks for reporting it!

**Comment by sksar** [comment]
> @fitrarhm @sksar @sweety-cool It seems to be an issue with the `--seed` argument of the command. Though, the Medusa project should have successfully been created, so you can navigate to the folder and run `yarn seed` as a workaround for now.
> 
> I'll investigate the issue. Thanks for reporting it!

The error was being caused even if we do not provide the seed. Then I looked into the logs, and in my case, it was a python not found.
In someplace of the install / create script, python is invoked from the shell `/bin/sh`.  I had python installed in my system, but as we know python2 is removed from homebrew, so the normal python from the command line does not work. It is now explicitly python3.
Invoking python3 would work, but the problem is that the command in the medusa create/install script is `python`.
So I tried to trick it by creating an alias in my .zshrc profile file. An alias of python to python3. 
But then again the problem was it just worked in zsh, because only zsh shell reads the .zshrc profile, but the command in the medusa install script is running from `/bin/sh`. So I tried to put the alias in some profile file which is read by `/bin/sh`. But nothing I did made it work.

So instead I thought to trick it in a way that a python bin file would be there, which would actually be a symlink and point to python3.
So I did this for my system
```
sudo ln -s -f /opt/homebrew/bin/python3 /usr/local/bin/python
```
thereby creating a symlink where python would normally go, (as of python2) and pointing it to where my python3 is installed. 
After doing this it worked.

So as per my understanding, in the medusa install script if there would have been a check to ensure if python is installed, or to use the python3 version by default, then this would have not been an issue for me. 

<hr/>

## So to summarize what problem I had and what I did:

- Medusa install script was failing, in the logs, it was showing `/bin/sh: python not found`
- Python 2 was not installed python3 was installed (from homebrew)
- python command does not work system-wide, python3 works.
- I tried creating an alias in my shell's profile file. Did not work as in medusa it was `/bin/sh`
- I tried creating a symlink from the usual place of python binary, pointing it to python3 using this `sudo ln -s -f /opt/homebrew/bin/python3 /usr/local/bin/python`
- This worked, because after doing this python command would point to python3 system-wide (no need to create aliases for different shells)

<hr>

<b>Hope this helps !</b>
<br/>

**Comment by kelvinnmugendi** [comment]
I think this should be reopened. I have been unable to install medusa for close to two days and just now found this Github issue. 
I get error 
```
npm ERR! code 1
npm ERR! path /Users/user/Documents/GitHub/my-medusa-store/node_modules/@stdlib/number-float64-base-normalize
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@10.0.1
npm ERR! gyp info using node@21.6.2 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.11.5 found at "/opt/homebrew/opt/python@3.11/bin/python3.11"
npm ERR! gyp info spawn /opt/homebrew/opt/python@3.11/bin/python3.11
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args 'binding.gyp',
npm ERR! gyp info spawn args '-f',
npm ERR! gyp info spawn args 'make',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/Users/user/Documents/GitHub/my-medusa-store/node_modules/@stdlib/number-float64-base-normalize/build/config.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/Users/user/Library/Caches/node-gyp/21.6.2/include/node/common.gypi',
npm ERR! gyp info spawn args '-Dlibrary=shared_library',
npm ERR! gyp info spawn args '-Dvisibility=default',
npm ERR! gyp info spawn args '-Dnode_root_dir=/Users/user/Library/Caches/node-gyp/21.6.2',
npm ERR! gyp info spawn args '-Dnode_gyp_dir=/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args '-Dnode_lib_file=/Users/user/Library/Caches/node-gyp/21.6.2/<(target_arch)/node.lib',
npm ERR! gyp info spawn args '-Dmodule_root_dir=/Users/user/Documents/GitHub/my-medusa-store/node_modules/@stdlib/number-float64-base-normalize',
npm ERR! gyp info spawn args '-Dnode_engine=v8',
npm ERR! gyp info spawn args '--depth=.',
npm ERR! gyp info spawn args '--no-parallel',
npm ERR! gyp info spawn args '--generator-output',
npm ERR! gyp info spawn args 'build',
npm ERR! gyp info spawn args '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp: binding.gyp not found (cwd: /Users/user/Documents/GitHub/my-medusa-store/node_modules/@stdlib/number-float64-base-normalize) while trying to load binding.gyp
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack at ChildProcess.<anonymous> (/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:271:18)
npm ERR! gyp ERR! stack at ChildProcess.emit (node:events:519:28)
npm ERR! gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:294:12)
npm ERR! gyp ERR! System Darwin 23.3.0
npm ERR! gyp ERR! command "/opt/homebrew/Cellar/node/21.6.2/bin/node" "/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /Users/user/Documents/GitHub/my-medusa-store/node_modules/@stdlib/number-float64-base-normalize
npm ERR! gyp ERR! node -v v21.6.2
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /Users/user/.npm/_logs/2024-02-19T08_32_13_796Z-debug-0.log
Error: Command failed with exit code 1: npm install
    at makeError (/opt/homebrew/lib/node_modules/@medusajs/medusa-cli/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/opt/homebrew/lib/node_modules/@medusajs/medusa-cli/node_modules/execa/index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async install (/opt/homebrew/lib/node_modules/@medusajs/medusa-cli/dist/commands/new.js:113:13)
    at async clone (/opt/homebrew/lib/node_modules/@medusajs/medusa-cli/dist/commands/new.js:173:5)
    at async newStarter (/opt/homebrew/lib/node_modules/@medusajs/medusa-cli/dist/commands/new.js:486:9) {
  shortMessage: 'Command failed with exit code 1: npm install',
  command: 'npm install',
  escapedCommand: 'npm install',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

```

**Comment by kelvinnmugendi** [comment]
Also, I think the issue should be named something along the lines of "Error: Command failed with exit code 1: npm install: MedusaJs" to help others search and locate the issue faster.

**Comment by kelvinnmugendi** [comment]
Also @olivermrbl can you take a quick look at this [question](https://stackoverflow.com/questions/78015331/how-to-fix-error-terminating-connection-due-to-administrator-command-vercel-p/78015594) on StackOverflow
Basically, am blocked from all entry points into MedusaJs, none of the project creation commands on the official site have worked for me
